### PR TITLE
Fix linters and ignore web tests for vm only tests

### LINKED ISSUE #507: Remove`pedantic` package
The `pedantic` package has been deprecated.

See https://pub.dev/packages/pedantic
It should be superseded by lints (Dart) or flutter_lints (Flutter).

### LINKED ISSUE #860: Support Flutter 3.x
`sdk: '>=2.12.0 <3.0.0'` has to be updated.
We should also test the basic features of the SDK, it should not be an issue because our tests run on the beta channel as well.

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
_#skip-changelog_

Dart 2.17 and Flutter 3.0 have new linters that needed to be fixed.
Added the new linter libs instead of the old ones.
Vm only tests are not run on web anymore.
Excluded windows from running web tests, since it was flaky.

## :bulb: Motivation and Context
Closes https://github.com/getsentry/sentry-dart/issues/507
Closes https://github.com/getsentry/sentry-dart/issues/860

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: dart/lib/src/sentry.dart [hunk 1]
```diff
@@ -105,8 +105,8 @@ class Sentry {
     // execute integrations after hub being enabled
     if (appRunner != null) {
       var runIntegrationsAndAppRunner = () async {
-        final integrations = options.integrations
-            .where((i) => !(i is RunZonedGuardedIntegration));
+        final integrations =
```
Comment (marandaneto) [inline dart/lib/src/sentry.dart:108]: @ueman I know you don't like it, but I actually find it easier to read when there many `({})` around.

File: dart/lib/src/transport/data_category.dart [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /// Different category types of data sent to Sentry. Used for rate limiting and client reports.
 enum DataCategory {
   all,
-  data_category_default, // default
+  dataCategoryDefault, // default
```
Comment (marandaneto) [inline dart/lib/src/transport/data_category.dart:4]: @denrase I hope this is fine, since camelCase is recommended by the new linter.

**Comment by marandaneto** [comment]
`webdev` has an issue https://github.com/dart-lang/webdev/issues/1607 and that's why it fails.

File: dart/lib/src/sentry.dart (hunk 1 above)
Comment (ueman) [reply dart/lib/src/sentry.dart:108]: Well, it's not my decision anymore 😅

File: dart/analysis_options.yaml [hunk 3]
```diff
@@ -18,3 +18,6 @@ linter:
   rules:
     prefer_relative_imports: true
     unnecessary_brace_in_string_interps: true
+    prefer_function_declarations_over_variables: false
+    no_leading_underscores_for_local_identifiers: false
+    avoid_renaming_method_parameters: false
```
Comment (ueman) [inline dart/analysis_options.yaml:23]: Maybe you want to enable this unnecessary string interpolation lint, too?

File: .github/workflows/flutter.yml [hunk 4]
```diff
@@ -77,7 +77,7 @@ jobs:
             flutter pub get
 
         - name: Test chrome
-          if: runner.os != 'macOS'
+          if: runner.os == 'Linux'
```
Comment (marandaneto) [inline .github/workflows/flutter.yml:80]: https://github.com/getsentry/sentry-dart/issues/500
Right now it only happens on Windows, let's see if that helps.

File: dart/lib/src/sentry.dart (hunk 1 above)
Comment (ueman) [reply dart/lib/src/sentry.dart:108]: Yeah, `is` is tricky when it comes to `!`.
Example:
```dart
class A {}
final a = A();

print(a is A); // true
print(a !is A); // true
print(a is! A); // false
```

File: dart/analysis_options.yaml (hunk 3 above)
Comment (marandaneto) [reply dart/analysis_options.yaml:23]: its enabled by default.

File: dart/lib/src/sentry.dart (hunk 1 above)
Comment (marandaneto) [reply dart/lib/src/sentry.dart:108]: Yeah, we only use `is!` and `is`, maybe that's why it's easy for me, so they logically make sense.
If we start using `!is`, it'd be confusing.

Comment (marandaneto) [reply dart/lib/src/sentry.dart:108]: > Well, it's not my decision anymore 😅

Your opinion counts :)

File: .github/workflows/dart.yml [hunk 5]
```diff
@@ -39,10 +39,10 @@ jobs:
             dart pub get
             dart test -p chrome --test-randomize-ordering-seed=random --chain-stack-traces
             dart test -p vm --coverage=coverage --test-randomize-ordering-seed=random --chain-stack-traces
-            dart pub run coverage:format_coverage --lcov --in=coverage --out=coverage/lcov.info --packages=.packages --report-on=lib
+            dart pub run coverage:format_coverage --lcov --in=coverage --out=coverage/lcov.info --packages=.dart_tool/package_config.json --report-on=lib
 
         - name: Install webdev
-          run: dart pub global activate webdev
+          run: dart pub global activate webdev 2.7.7
```
Comment (marandaneto) [inline .github/workflows/dart.yml]:
```suggestion
          run: dart pub global activate webdev
```

**Comment by marandaneto** [comment]
> `webdev` has an issue [dart-lang/webdev#1607](https://github.com/dart-lang/webdev/issues/1607) and that's why it fails.

They fixed it.

**Review by brustolin — APPROVED** [review]
LGTM!

**Pull request merged**